partitioner-config-service-impl/src/main/java/org/hypertrace/partitioner/config/service/PartitionerConfigServiceImpl.java

```diff
@@ -0,0 +1,86 @@
+package org.hypertrace.partitioner.config.service;
+
+import com.google.inject.Inject;
+import io.grpc.stub.StreamObserver;
+import java.util.List;
+import java.util.Optional;
+import lombok.extern.slf4j.Slf4j;
+import org.hypertrace.partitioner.config.service.store.PartitionerProfilesStore;
+import org.hypertrace.partitioner.config.service.v1.*;
+
+@Slf4j
+public class PartitionerConfigServiceImpl
+    extends PartitionerConfigServiceGrpc.PartitionerConfigServiceImplBase {
+
+  private final PartitionerProfilesStore partitionerProfilesStore;
+  private final PartitionerConfigServiceRequestValidator validator;
+
+  @Inject
+  public PartitionerConfigServiceImpl(
+      PartitionerProfilesStore partitionerProfilesStore,
+      PartitionerConfigServiceRequestValidator validator) {
+    this.partitionerProfilesStore = partitionerProfilesStore;
+    this.validator = validator;
+  }
+
+  public void getPartitionerProfile(
+      GetPartitionerProfileRequest request,
+      StreamObserver<GetPartitionerProfileResponse> responseObserver) {
+    try {
+      validator.validateOrThrow(request);
+      Optional<PartitionerProfile> profile =
+          partitionerProfilesStore.getPartitionerProfile(request.getProfile());
+      if (profile.isPresent()) {
```

Comment:
should throw Status.NOT_FOUND (or whatever it's called) if not present, since this is effectively a get by id.

Reply:
fixed